### ISSUE TITLE
🐛 `migrate --write` does not respect `biome.json`'s formatting config while rewriting it (`beta.1`)

### ISSUE THREAD

**Issue by lgarron** [biomejs/biome#5770]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v23.11.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.9

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /Users/lgarron/Code/git/github.com/cubing/cubing.js/biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

Repro (uses `bun`):

```shell
git clone https://github.com/cubing/cubing.js && cd cubing.js
git checkout 1196be4cb1fb88ab5153adb8414ae64efa31c978
bun add @biomejs/biome@beta
bun x @biomejs/biome@beta migrate --write
```

Observe that `biome.json` uses tabs for indentation.

### Expected result

As specified in the config file being analyzed and rewritten, indentation should use two spaces.

(At least `bun x @biomejs/biome@beta check --write 'biome.json'` works, though.)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
This should be easy to fix. We should use this code 

```rs
                    workspace.change_file(ChangeFileParams {
                        project_key,
                        path: biome_path.clone(),
                        content: new_configuration_content,
                        version: 1,
                    })?;
                    let printed = workspace.format_file(FormatFileParams {
                        project_key,
                        path: biome_path,
                    })?;

```

Instead of this 
https://github.com/biomejs/biome/blob/b675840b32e3c8c5514542c6f67c5ca4890c16f2/crates/biome_cli/src/execute/migrate.rs#L289-L300
